R/utils.R

```diff
@@ -125,3 +126,72 @@ get_api_results <- function(file_id) {
 
   }
 }
+
+#' Does x match the pattern of a delimited data file path?
+#' @noRd
+is_delim_path <- function(path) {
+  grepl("\\.(csv|tsv)", path)
+}
+
+#' Does x match the pattern of a URL?
+#' @noRd
+is_url <- function(x) {
+  if (!is_vector(x) || is_empty(x)) {
+    return(FALSE)
+  }
+
+  url_pattern <-
+    "http[s]?://(?:[[:alnum:]]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+
+  grepl(url_pattern, x)
+}
+
+#' Convert a sf object to a data frame of coordinates
+#'
+#' @noRd
+#' @importFrom purrr list_cbind
+convert_to_coords <- function(data,
```

Comment:
We had a concern about this added functionality leading to non-point spatial data being used/interpreted improperly in the tool. For example, if someone uploaded a dataset of parks with multipolygon geometries, a park that spans multiple census tracts would be assigned a single point via the `st_centroid` or `st_point_on_surface` function. On the back-end, our tool would then "attribute" the park only to the census tract that the single point falls into for purposes of the disparity calculations. That wouldn't be quite correct, since in reality the park falls in multiple census tracts (and you'd likely want to weight the assignment by area in each tract).  

We've heard from other folks about the desire to use non-point spatial data with the tool and we're about to pilot some use cases with one of our partners to determine where/how to use these data properly. Until we have a "best practice" that we're comfortable implementing as a default in sedtR, we'd rather individuals have to make the choice about how to preprocess their data to convert to points before using sedtR

Reply:
Totally fair. I think the scenario I had in mind was a case when you are trying to count parks by county or parcels by neighborhood but you could always supply an example showing how to use `sf::st_centroid()` to transform data in advance of passing it to the API. If you want to revisit this in the future, I can imagine including some warning similar to the default warning for `sf::st_centroid()` ("st_centroid assumes attributes are constant over geometries") or  some check that compares the input geometry of the resource to the comparison geometry. Not sure if that is possible or desirable though. I'll plan to revise the PR to pull this out.